app/human-detector-app/auth/KeyCloakAuth.ts

```diff
@@ -0,0 +1,92 @@
+import { makeRedirectUri, useAuthRequest } from 'expo-auth-session';
+import * as AuthSession from 'expo-auth-session';
+import * as CryptoJS from 'crypto-js';
+
+function genCodeVerifier(size: number) {
+  const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890';
+  const charLength = chars.length;
+  let result = '';
+  for (let i = 0; i < size; i++) {
+    result += chars.charAt(Math.floor(Math.random() * charLength));
+  }
+  return result;
+}
+
+/**
+ * KeyCloak instance class to hold a KeyCloak auth instance. This object
+ * will hold values that will be important to keep track of.
+ *
+ * To get the authorization token, we need to make sure that codeVerifier and
+ * redirectUri stay the same.  These are used for verification of authorization.
+ */
+export default class KeyCloakInstance {
+  // Code verifier to be sent
+  codeVerifier: string = '';
+
+  redirectUri: string;
+
+  constructor() {
+    this.codeVerifier = genCodeVerifier(100);
```

Comment:
This should be regenerated every time we request a new authorization code

Reply:
Does it need to be regenerated every time?

Reply:
[The spec](https://stackoverflow.com/a/58845905) (:nerd_face:) says that we should generate a unique code verifier for each authorization code request, so we should probably do that. Generating once might be fine for now because it's S256 hashed, but generating a new one for each request is easy enough

Reply:
discussed offline: cafe33972623d395fa57aac25ff70a9317d41029